x/params/types/subspace.go

```diff
@@ -147,19 +147,15 @@ func (s Subspace) Modified(ctx sdk.Context, key []byte) bool {
 }
 
 // checkType verifies that the provided key and value are comptable and registered.
-func (s Subspace) checkType(key []byte, value interface{}) {
+func (s Subspace) checkType(key []byte, value codec.ProtoMarshaler) {
 	attr, ok := s.table.m[string(key)]
 	if !ok {
 		panic(fmt.Sprintf("parameter %s not registered", string(key)))
 	}
 
-	ty := attr.ty
-	pty := reflect.TypeOf(value)
-	if pty.Kind() == reflect.Ptr {
-		pty = pty.Elem()
-	}
+	registeredProtoMarshaler := attr.ty
 
-	if pty != ty {
+	if registeredProtoMarshaler != value {
```

Comment:
This actually needs to check for type equality, not value equality.

Is there a way to do that without reflection?

Reply:
You mean the concrete types? No, I can't think of any other way apart from reflection.